lib/plugins/aws/package/compile/events/schedule.js

```diff
@@ -148,74 +195,170 @@ class AwsCompileScheduledEvents {
+                const scheduleLogicalId = this.provider.naming.getScheduleLogicalId(
+                  functionName,
+                  scheduleNumberInFunction
+                );
+
+                const lambdaPermissionLogicalId =
+                  this.provider.naming.getLambdaSchedulePermissionLogicalId(
+                    functionName,
+                    scheduleNumberInFunction
+                  );
 
-              _.merge(
-                this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
-                newScheduleObject,
-                newPermissionObject
-              );
+                let templateScheduleExpression;
+                if (typeof ScheduleExpression === 'string') {
+                  templateScheduleExpression = `"${ScheduleExpression}"`;
+                } else {
+                  templateScheduleExpression = JSON.stringify(ScheduleExpression);
+                }
+
+                const scheduleTemplate = `
+                  {
+                    "Type": "AWS::Events::Rule",
+                    ${dependsOn ? `"DependsOn": "${dependsOn}",` : ''}
+                    "Properties": {
+                      "ScheduleExpression": ${templateScheduleExpression},
+                      "State": "${State}",
+                      ${Name ? `"Name": "${Name}",` : ''}
+                      ${Description ? `"Description": "${Description}",` : ''}
+                      "Targets": [{
+                        ${Input ? `"Input": "${Input}",` : ''}
+                        ${InputPath ? `"InputPath": "${InputPath}",` : ''}
+                        ${InputTransformer ? `"InputTransformer": ${InputTransformer},` : ''}
+                        "Arn": ${lambdaTargetJson},
+                        "Id": "${scheduleId}"
+                      }]
+                    }
+                  }
+                `;
+
+                const permissionTemplate = `
+                  {
+                    "Type": "AWS::Lambda::Permission",
+                    ${dependsOn ? `"DependsOn": "${dependsOn}",` : ''}
+                    "Properties": {
+                      "FunctionName": ${lambdaTargetJson},
+                      "Action": "lambda:InvokeFunction",
+                      "Principal": "events.amazonaws.com",
+                      "SourceArn": { "Fn::GetAtt": ["${scheduleLogicalId}", "Arn"] }
+                    }
+                  }
+                `;
+
+                const newScheduleObject = {
+                  [scheduleLogicalId]: JSON.parse(scheduleTemplate),
+                };
+
+                const newPermissionObject = {
+                  [lambdaPermissionLogicalId]: JSON.parse(permissionTemplate),
+                };
+
+                Object.assign(resources, newScheduleObject, newPermissionObject);
+              }
             }
           }
         });
       }
     });
+
+    if (hasSchedulerEvents) {
+      if (!resources.IamRoleLambdaExecution) {
+        log.notice(
```

Comment:
Let's make it `info` level (so it appears only in verbose mode - otherwise it feels a bit too noisy)